pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/BulkProcessor.java

```diff
@@ -249,6 +248,7 @@ public void execute(final BulkRequest bulkRequest, final long executionId) {
             try {
                 this.semaphore.acquire();
             } catch (InterruptedException ex) {
+                this.semaphore.release();
```

Comment:
I am really not sure about this.
if `acquire` fails you are supposed to not have 'acquired' it